doc/source/workflows/events.rst

```diff
@@ -38,6 +39,47 @@ Workflow events are a special type of workflow task. They "finish" when the even
     workflow.run(gather.bind(event1_task, event2_task))
 
 
+HTTP Events
+-----------
+
+:ref:`Ray Serve <rayserve>` enables receiving workflow events via HTTP.
+To use HTTP events, we will install an HTTP event listener in the workflow and use Ray Serve to expose an HTTP endpoint. Below is an end-to-end example for installing the endpoint and receiving an HTTP event in a workflow.
+
+
+To listen for HTTP events in a workflow, we will use an ``HTTPListener``. Each ``HTTPListener`` subscribes to a unique `workflow_id` and `event_key` pair. To send an event to the listener, an HTTP request from
+an external client should specify ``workflow_id`` as part of the request URL and the ``event_key`` and ``event_payload`` keys in the JSON request body (see below).
+
+.. literalinclude:: ./doc_code/wait_for_event_http.py
+   :language: python
+   :start-after: __wait_for_event_begin__
+   :end-before: __wait_for_event_end__
+
+To create an HTTP endpoint through which we can send events to the workflow, we will use :ref:`Ray Serve <rayserve>` to deploy an ``HTTPEventProvider``.
+This will create an endpoint at ``http://hostname:port/event/send_event/<workflow_id>``. If Ray Serve is deployed locally,
+the endpoint may be reached at ``http://127.0.0.1:8000/event/send_event/<workflow_id>``.
+Note that to send an HTTP event to a workflow event listener, the HTTP request must include the same ``workflow_id``.
+Each request should also include a JSON body with two fields: ``event_key`` and ``event_payload``, as shown in the example
+below. The ``event_key`` field should match the argument passed to ``workflow.wait_for_event`` on the listener side. In the workflow, once an HTTP event is received, the event task will return the value of the ``event_payload`` field.
+
+In summary, to trigger an HTTP event in the workflow, an external client should have:
+
+* the HTTP endpoint address (e.g. `http://127.0.0.1:8000/event/send_event/`)
+* the ``workflow_id`` (e.g. "workflow_receive_event_by_http")
+* a valid JSON formatted message with the fields ``event_key`` and ``event_payload``, where ``event_key`` matches the one used in the workflow
+
+The HTTP request will receive a reply once the event has been received by the workflow. The returned status code can be:
+
+1. 404: either ``workflow_id`` or ``event_key`` cannot be found.
```

Comment:
We should explain in which case this might happen. (ordering matters here)

Reply:
Done. Status code reordered.